op-supervisor/supervisor/backend/reads/full.go

```diff
@@ -0,0 +1,191 @@
+package reads
```

Comment:
nit: I might quibble a bit that we have `query.go` for `db` but then also separately `reads`. Renaming this `read_safety` or something might be better. IDK, not worth changing at this point.